components/builder-api/src/server/handlers.rs

```diff
@@ -223,6 +225,80 @@ pub fn get_profile(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+pub fn get_access_tokens(req: &mut Request) -> IronResult<Response> {
+    let session_id = {
+        let session = req.extensions.get::<Authenticated>().unwrap();
+        session.get_id()
+    };
+
+    let mut request = AccountTokensGet::new();
+    request.set_account_id(session_id);
+    request.set_revoked(false);
+
+    match route_message::<AccountTokensGet, AccountTokens>(req, &request) {
+        Ok(account_tokens) => Ok(render_json(status::Ok, &account_tokens)),
+        Err(err) => return Ok(render_net_error(&err)),
+    }
+}
```

Comment:
I'm probably missing something, but the presence of this function confuses me a little.  Don't you need to be authenticated to call this function?  And being authenticated would necessitate the use of an access token?  It feels like a chicken/egg problem.

Reply:
Yeah its a bit confusing - we have multiple ways to auth.  The web auth flow is oauth -> session tokens.  That is not changing currently.  The personal access tokens are targeted to the hab cli (and build worker) flow.  In order for a user to generate/revoke their access token, they need to come to our web site.  Therefore, they will need to auth there first, and that is totally what we want.